Detectors/FIT/FT0/simulation/src/Digits2Raw.cxx

```diff
@@ -104,19 +119,6 @@ void Digits2Raw::readDigits(const std::string fileDigitsName)
       if (nHBF) {
         for (int j = 0; j < nHBF - 1; j++) {
           o2::InteractionRecord rdhIR = HBIRVec[j];
-          for (int link = 0; link < (int)mPages.size(); ++link) {
-            setRDH(mPages[link].mRDH, link, rdhIR);
-            mPages[link].flush(mFileDest);
-          }
-        }
-
-        uint32_t current_orbit = intRecord.orbit;
-        if (old_orbit != current_orbit) {
-          for (DataPageWriter& writer : mPages)
-            writer.flush(mFileDest);
-          for (int nlink = 0; nlink < NPMs; ++nlink)
-            setRDH(mPages[nlink].mRDH, nlink, intRecord);
-          old_orbit = current_orbit;
         }
         auto channels = bcd.getBunchChannelData(digitsCh);
```

Comment:
I don't understand the line 117-128 but they look wrong. 1st of all, with RawWriter, unless you want to have some special treatment for the HBFs w/o any signal (e.g. add some detector-specific info), you don't need to deal with HBFUtils at all.
Then, on 120-122 you have dummy loop which just assigns a loop-local variable. I suspect the code should be simply:
````
for (int ibc = 0; ibc < nbc; ibc++) {
  auto& bcd = digitsBC[ibc];
  intRecord = bcd.getIntRecord();
  auto channels = bcd.getBunchChannelData(digitsCh);
  if (!channels.empty()) ;
      convertDigits(bcd, channels, lut, intRecord);
  }
}
````

Reply:
I think this  structure sets similar IntRecord for all detector. 
int nHBF = mSampler.fillHBIRvector(HBIRVec, lastIR, intRecord);
        if (nHBF) {
        for (int j = 0; j < nHBF - 1; j++) {
          o2::InteractionRecord rdhIR = HBIRVec[j];
 I can remove it, no problem